lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php

```diff
@@ -121,39 +122,64 @@ public function convertException($message, DriverException $exception)
         return new Exception\DriverException($message, $exception);
     }
 
+
     /**
      * {@inheritdoc}
      */
     public function createDatabasePlatformForVersion($version)
     {
-        if ( ! preg_match('/^(?P<major>\d+)(?:\.(?P<minor>\d+)(?:\.(?P<patch>\d+))?)?/', $version, $versionParts)) {
-            throw DBALException::invalidPlatformVersionSpecified(
-                $version,
-                '<major_version>.<minor_version>.<patch_version>'
-            );
-        }
+        // Detect vendor and version
 
         if (false !== stripos($version, 'mariadb')) {
-            return $this->getDatabasePlatform();
-        }
 
-        $majorVersion = $versionParts['major'];
-        $minorVersion = isset($versionParts['minor']) ? $versionParts['minor'] : 0;
-        $patchVersion = isset($versionParts['patch']) ? $versionParts['patch'] : null;
+            // Hack to ignore things like '5.5.5-10.2.8-MariaDB-10.2.8+maria~xenial-log'
+            $version = str_replace('5.5.5-', '', $version);
 
-        if ('5' === $majorVersion && '7' === $minorVersion && null === $patchVersion) {
-            $patchVersion = '9';
-        }
+            if (!preg_match('/^(mariadb-)?(?P<major>\d+)\.(?P<minor>\d+)\.(?P<patch>\d+)/', strtolower($version), $versionParts)) {
+                throw DBALException::invalidPlatformVersionSpecified(
+                    $version,
+                    '(mariadb-)?<major_version>.<minor_version>.<patch_version>'
+                );
+            }
+
+            $version = $versionParts['major'] . '.' . $versionParts['minor'] . '.' . $versionParts['patch'];
+
+            if (version_compare($version, '10.2.7', '>=')) {
+                return new MariaDb102Platform();
+            }
+
+        } else {
+            // Oracle MySQL
 
-        $version = $majorVersion . '.' . $minorVersion . '.' . $patchVersion;
+            if (!preg_match('/^(?P<major>\d+)(?:\.(?P<minor>\d+)(?:\.(?P<patch>\d+))?)?/', $version, $versionParts)) {
+                throw DBALException::invalidPlatformVersionSpecified(
+                    $version,
+                    '<major_version>.<minor_version>.<patch_version>'
+                );
+            }
+            $majorVersion = $versionParts['major'];
+            $minorVersion = isset($versionParts['minor']) ? $versionParts['minor'] : 0;
+            $patchVersion = isset($versionParts['patch']) ? $versionParts['patch'] : null;
 
-        if (version_compare($version, '5.7.9', '>=')) {
-            return new MySQL57Platform();
+            if ('5' === $majorVersion && '7' === $minorVersion && null === $patchVersion) {
+                $patchVersion = '9';
+            }
+
+            $version = $majorVersion . '.' . $minorVersion . '.' . $patchVersion;
+
+            if (version_compare($version, '5.7.9', '>=')) {
+                return new MySQL57Platform();
+            }
         }
 
+
```

Comment:
Excessive spacing

Reply:
I'll fix.. If you're alright with this. I propose to use `php-cs-fixer fix --rules @PSR2 ./lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php` (also for all the files modified by this P/R). If PSR-2 is not wanted, let me know what standard to use, I could not find a code styling config (php-cs-fixer or phpcbf) ?